WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -76,11 +77,42 @@ private void StepInputRegistrationPhase()
 				}
 				else
 				{
-					round.SetPhase(Phase.ConnectionConfirmation);
+					var thereAreOffendingAlices = false;
+					await foreach (var offendingAlices in CheckTxoSpendStatusAsync(round).WithCancellation(cancel).ConfigureAwait(false))
+					{
+						if (offendingAlices.Any())
+						{
+							thereAreOffendingAlices = true;
+							round.Alices.RemoveAll(x => offendingAlices.Contains(x));
+						}
+					}
+					if (!thereAreOffendingAlices)
+					{
+						round.SetPhase(Phase.ConnectionConfirmation);
+					}
```

Comment:
```suggestion
					round.SetPhase(Phase.ConnectionConfirmation);
```

Reply:
Wait, this doesn't look right to me, we don't know how many alices remain in the round. In the current version there is a minimum number of alices required to proceed, otherwise the round is aborted. We can do something like that or don't move to connection confirmation.

Reply:
yes, you're right, this should be conditional. if the reason it's being checked is a timeout, fail. if it's because the max number max number of inputs was reached, but then one of them turns out to be double spent, just deregister it and do nothing for now. but it should not directly fail the round unless necessary, and sometimes it's not necessary at this phase